src/core/initializeComponentsFactory.js

```diff
@@ -45,13 +45,13 @@ export default (
   });
   config.validate();
   componentCreators.forEach(createComponent => {
-    const { namespace } = createComponent;
+    const { namespace, cookieNamespace } = createComponent;
     // TO-DOCUMENT: Helpers that we inject into factories.
     let component;
     try {
       component = createComponent({
         logger: logger.spawn(`[${namespace}]`),
-        cookie: createCookie(cookieProxy, namespace),
+        cookie: createCookie(cookieProxy, cookieNamespace || namespace),
```

Comment:
I'm a bit concerned that one of the components that doesn't have `cookieNamespace` defined (or some new component) will start using the cookie utility without the developer realizing they need to define `cookieNamespace` and we won't notice until the library in prod and we'll be locked into using `namespace` for that component forever instead of `cookieNamespace`. We could throw an error here or in createCookie but I think it would also be reasonable to just add `cookieNamespace` to all the components right now, which I believe would greatly reduce the chance of this happening.